test/test_device_reduce.cu

```diff
@@ -521,91 +544,188 @@ cudaError_t Dispatch(
+                                  EndOffsetIteratorT   d_segment_end_offsets,
+                                  ReductionOpT         reduction_op,
+                                  bool                 debug_synchronous)
 {
-#ifndef CUB_CDP
-    (void)timing_iterations;
-    (void)d_temp_storage_bytes;
-    (void)d_cdp_error;
-    (void)d_temp_storage;
-    (void)temp_storage_bytes;
-    (void)d_in;
-    (void)d_out;
-    (void)num_items;
-    (void)max_segments;
-    (void)d_segment_begin_offsets;
-    (void)d_segment_end_offsets;
-    (void)reduction_op;
-    (void)debug_synchronous;
-    *d_cdp_error = cudaErrorNotSupported;
-#else
-    *d_cdp_error = Dispatch(Int2Type<CUB>(), timing_iterations, d_temp_storage_bytes, d_cdp_error, d_temp_storage, temp_storage_bytes,
-        d_in, d_out, num_items, max_segments, d_segment_begin_offsets, d_segment_end_offsets, reduction_op, 0, debug_synchronous);
-    *d_temp_storage_bytes = temp_storage_bytes;
-#endif
+  *d_cdp_error = Dispatch(cub_backend,
+                          timing_iterations,
+                          d_temp_storage_bytes,
+                          d_cdp_error,
+                          d_temp_storage,
+                          temp_storage_bytes,
+                          d_in,
+                          d_out,
+                          num_items,
+                          max_segments,
+                          d_segment_begin_offsets,
+                          d_segment_end_offsets,
+                          reduction_op,
+                          0,
+                          debug_synchronous);
+
+  *d_temp_storage_bytes = temp_storage_bytes;
 }
 
-
 /**
- * Dispatch to CUB_CDP kernel
+ * Launch kernel and dispatch on device. Should only be called from host code.
+ * The CubBackend should be one of the non-CDP CUB backends to invoke from the
+ * device.
  */
-template <typename InputIteratorT, typename OutputIteratorT, typename BeginOffsetIteratorT, typename EndOffsetIteratorT, typename ReductionOpT>
-CUB_RUNTIME_FUNCTION __forceinline__
-cudaError_t Dispatch(
-    Int2Type<CUB_CDP>       dispatch_to,
-    int                 timing_iterations,
-    size_t              *d_temp_storage_bytes,
-    cudaError_t         *d_cdp_error,
-
-    void*               d_temp_storage,
-    size_t&             temp_storage_bytes,
-    InputIteratorT      d_in,
-    OutputIteratorT     d_out,
-    int                 num_items,
-    int                 max_segments,
-    BeginOffsetIteratorT d_segment_begin_offsets,
-    EndOffsetIteratorT  d_segment_end_offsets,
-    ReductionOpT        reduction_op,
-    cudaStream_t        stream,
-    bool                debug_synchronous)
+template <int CubBackend,
+          typename InputIteratorT,
+          typename OutputIteratorT,
+          typename BeginOffsetIteratorT,
+          typename EndOffsetIteratorT,
+          typename ReductionOpT>
+CUB_RUNTIME_FUNCTION cudaError_t
```

Comment:
Same question as above. If there's something about marking `LaunchCDPKernel` as `CUB_RUNTIME_FUNCTION` that I'm missing, please ignore this. Otherwise, fix other places below as well.